### PR TITLE
SDIT-1349 Add Activities prison feature switch check to UI

### PR DESCRIPTION
**Pull request by mikehalmamoj**
The new feature is **DPS feature switch not turned on for WWI**.

(I'll try and sort out that horrible formatting of the `Activate Prison` button soon)

image

### CODE REVIEW

File: server/routes/activitiesMigration/activitiesMigrationController.ts [hunk 1]
```diff
@@ -102,13 +104,25 @@ export default class ActivitiesMigrationController {
...
       req.session.startActivitiesMigrationForm.dlqCount = dlqCount.toLocaleString()
       req.session.startActivitiesMigrationForm.incentiveLevelIds = incentiveLevels.map(
         (level: IncentiveLevel) => level.code,
       )
-      req.session.startActivitiesMigrationForm.prisonSwitchedOn = featureSwitchOn
+      req.session.startActivitiesMigrationForm.prisonSwitchedOnNomis = nomisFeatureSwitchOn
+      req.session.startActivitiesMigrationForm.prisonSwitchedOnDps =
+        dpsPrisonRollout === null ||
+        (dpsPrisonRollout.activitiesRolledOut &&
+          dpsPrisonRollout.activitiesRolloutDate <= moment().format('YYYY-MM-DD'))
```
Comment (mikehalmamoj) [inline server/routes/activitiesMigration/activitiesMigrationController.ts:125]: This is the main change - get the `PrisonRolloutPlan` for the prison and check if the prison is rolled out on or before today.

File: server/views/pages/activities/startActivitiesMigrationPreview.njk [hunk 2]
```diff
@@ -52,6 +52,11 @@
                                     }) }}
                                 </p>
                             {% endif %}
+                            {% if not form.prisonSwitchedOnDps %}
+                                <p class="govuk-notification-banner__heading" id="dpsFeatureSwitch">
+                                    DPS feature switch not turned on for {{ form.prisonId }}.
+                                </p>
+                            {% endif %}
```
Comment (mikehalmamoj) [inline server/views/pages/activities/startActivitiesMigrationPreview.njk:59]: Show the warning if the prison is not feature switched on in DPS.

File: server/services/activitiesService.ts [hunk 3]
```diff
@@ -2,28 +2,34 @@ import RestClient from '../data/restClient'
...
+  constructor(private readonly hmppsAuthClient: HmppsAuthClient) {}
+
   private static restClient(token: string): RestClient {
     return new RestClient('Activities API Client', config.apis.activities, token)
   }
 
   async getRolloutPrison(prisonId: string, context: Context): Promise<RolloutPrisonPlan> {
-    return ActivitiesService.restClient(context.token).get<RolloutPrisonPlan>({
+    const token = await this.hmppsAuthClient.getSystemClientToken(context.username)
+    return ActivitiesService.restClient(token).get<RolloutPrisonPlan>({
```
Comment (mikehalmamoj) [inline server/services/activitiesService.ts:16]: I got mxed up between user tokens and system tokens. As usual.

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**